neurodamus/ngv.py

```diff
@@ -81,99 +129,41 @@ def gid(self, val: int):
-        #            morph_wrap.section_index2name_dict[sec_index],
-        #            er_area,
-        #            er_vol)
-        #            for sec_index, er_area, er_vol in zip(
-        #            morph_wrap.morph.endoplasmic_reticulum.section_indices,
-        #            morph_wrap.morph.endoplasmic_reticulum.surface_areas,
-        #            morph_wrap.morph.endoplasmic_reticulum.volumes)))
-        return cmds
-
-    @staticmethod
-    def _secparams_as_hoc(_morph_wrap):
-        """Create hoc commands for section parameters (perimeters & cross-sectional area)
-        :param morph_wrap: MorphIOWrapper object holding MorphIO morphology object
-
-        For example:
-            dend[0] { perimeter_mcd = 32 cross_sectional_area_mcd = 33}
-        """
-        cmds = []
-        # these parameters will be used in the near future by the model but temporarily disabled
-        # cmds.extend(("{} {{ perimeter_mcd = {:g} cross_sectional_area_mcd = {:g} }}".format(
-        #     morph_wrap.section_index2name_dict[morph_sec_index + 1],
-        #     sec_perimeter,
-        #     sec_xsect_area)
-        #     for morph_sec_index, sec_perimeter, sec_xsect_area in
-        #     (Astrocyte._mcd_section_parameters(sec) for sec in morph_wrap.morph.sections)))
-        return cmds
-
-    def _show_mcd(sec):
-        if not hasattr(sec(0.5), "cadfifus"):
-            logging.info("No cadifus mechanism found")
-            return
-
-    # the following lines are useful for debugging
-    #        logging.info("{}: \tP={:.4g}\tX-Area={:.4g}\tER[area={:.4g}\tvol={:.4g}]".format(
-    #            sec,
-    #            sec(0.5).mcd.perimeter,
-    #            sec(0.5).mcd.cross_sectional_area,
-    #            sec(0.5).mcd.er_area,
-    #            sec(0.5).mcd.er_volume)
-    #        )
-
-    def set_pointers(self):
-        # the endfeet are not included in all as they are added later.
-        # I still do not know exactly when the pointers need to be
-        # reassigned and which ones are stale. the endfeet may be already
-        # up-to-date
-        # issue: https://github.com/openbraininstitute/neurodamus/issues/263
-        all_secs = chain(self._cellref.all, self.endfeet)
-
-        # just a safety check
-        assert len(self._cellref.all) + len(self.endfeet) == len(self.sections_glut), (
-            "Mismatch between sections and sections_glut: "
-            "probably some sections are unaccounted for"
-        )
-
-        for glut, sec in zip(self.sections_glut, all_secs):
-            Nd.setpointer(glut._ref_glut, "glu2", sec(0.5).cadifus)
+        for sec, parent_id, length, diameter, R0pas in zip(
+            self.endfeet, parent_ids, lengths, diameters, R0passes
+        ):
+            is_resized = self._init_endfoot_section(sec, parent_id, length, diameter, R0pas)
+            assert not is_resized, "Endfeet sections should be created with only 1 compartment"
```

Comment:
Do we assert this currently in the main branch? 
If it needs to raise errors, it seems efficient to assert earlier before the execution in `_init_endfoot_section` ?

Reply:
this should never happen. If you want I can remove the check. The thing is that for `all` the function returns if there were resizes and that is relevant. Here I just have the info: `is_resized` and should always be false.

Reply:
Then we can remove this check, and `_init_endfoot_section` takes `is_resized` from `_init_basic_section` but doesn't return it. What do you think?

Reply:
ok, np. Done